pinning/pinner/dspinner/pin.go

```diff
@@ -291,7 +335,14 @@ func (p *pinner) doPinDirect(ctx context.Context, c cid.Cid, name string) error
 		return err
 	}
 
-	return p.flushPins(ctx, false)
+	err = p.flushPins(ctx, false)
+	if err != nil {
+		return err
+	}
+	if p.rootsProvider != nil {
+		return p.rootsProvider.Provide(ctx, c, true)
```

Comment:
```suggestion
		go p.rootsProvider.Provide(ctx, c, true)
```
unless we want the pin operation to fail on provide failure.